### DISCUSSION TITLE
I have a problem

### DISCUSSION THREAD

**Discussion by R3B4T0** [lcobucci/jwt#741] [Q&A] [answered]
I have a problem when I try to install lcobucci/jwt, it returns: "Your requirements could not be resolved to an installable set of packages.
Problem 1
    - lcobucci/jwt[4.1.0, ..., 4.2.x-dev] require ext-sodium * -> it is missing from your system. Install or enable PHP's sodium extension.
    - Root composer.json requires lcobucci/jwt ^4.1 -> satisfiable by lcobucci/jwt[4.1.0, ..., 4.2.x-dev]."

**Comment by lcobucci** [comment, marked as answer]
You've got to make sure the sodium extension is enabled in your installation (as it's bundled in php it's commonly enabled by default - not in your case though).

https://www.php.net/manual/en/sodium.installation.php should give you some info but it's quite likely that you have a pre-compiled package for your OS.

**Reply by R3B4T0** [reply]
The extension was disabled whithin the php.ini

Thank you for your help!!!

**Comment by JMulungo-xvii** [comment]
I have got a similar issue trying to run "composer update" => Problem 1
    - Root composer.json requires lcobucci/jwt ^3.4 -> satisfiable by lcobucci/jwt[3.4.0, ..., 3.4.x-dev].
    - lcobucci/jwt[3.4.0, ..., 3.4.x-dev] require php ^5.6 || ^7.0 -> your php version (8.2.4) does not satisfy that requirement.

**Reply by lcobucci** [reply]
You need to upgrade to v4 or v5, which are compatible with php 8.x. Check our upgrading guides on the docs 👍

**Comment by Janam007** [comment]
- lcobucci/jwt 4.3.0 requires ext-sodium * -> it is missing from your system. Install or enable PHP's sodium extension.

i have php version 8.2.4 

warning: pecl/libsodium requires PHP (version >= 7.0.0, version <= 8.0.99), installed version is 8.2.4
downloading libsodium-2.0.23.tgz ...
 i am not bieng able to install libsodium in my system for my laravel project

**Reply by Ocramius** [reply]
Libsodium is part of PHP itself: https://github.com/php/php-src/tree/115c60e0bbe0dbe1c0b11db7bf50cf23695f14dd/ext/sodium

**Comment by Janam007** [comment]
Dear Sir,
 it is saying ext-sodium is missing from your system.


 Problem 1

    - lcobucci/jwt is locked to version 4.3.0 and an update of this package
was not requested.

    - lcobucci/jwt 4.3.0 requires ext-sodium * -> it is missing from your
system. Install or enable PHP's sodium extension.

  Problem 2

    - lcobucci/jwt 4.3.0 requires ext-sodium * -> it is missing from your
system. Install or enable PHP's sodium extension.

    - kreait/firebase-tokens 4.3.0 requires lcobucci/jwt ^4.3.0|^5.0 ->
satisfiable by lcobucci/jwt[4.3.0].

    - kreait/firebase-tokens is locked to version 4.3.0 and an update of
this package was not requested.


can you please help me no this ..

Best Regards,
Janam
ktm, Nepal

**Comment by SvenRtbg** [comment]
Even though the Sodium extension usually is installed by default, it isn't guaranteed to be installed. The documentation states it is still a compile time directive that enables it, and it is entirely possible that YOUR PHP version does not include it by default.

Verify which extensions are available: `phpinfo(INFO_MODULES)` should output something, as well as `print_r(get_loaded_extensions());`. Any missing "sodium" would proof your PHP is lacking support of it.

**Comment by Eradah-alfakih** [comment]
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - lcobucci/jwt 5.2.0 requires ext-sodium * -> it is missing from your system. Install or enable PHP's sodium extension. 
    -in cpanel

**Reply by lcobucci** [reply]
In theory cpanel does allow you to enable it: https://support.cpanel.net/hc/en-us/articles/360056786594-How-to-Install-the-Sodium-Cryptographic-Library-libsodium-and-PHP-Extension-on-CentOS-7-and-CloudLinux-7

You'd need to contact the support team of your hosting provider, in case you're not able to do so.

**Comment by 2earn** [comment]
I use laragon 👍 
i got this warning:

> Warning: PHP Startup: Unable to load dynamic library 'sodium' (tried: C:/laragon/bin/php/php-8.2.23-nts-Win32-vs16-x64/ext\sodium (The specified module could not be found), C:/laragon/bin/php/php-8.2.23-nts-Win32-vs16-x64/ext\php_sodium.dll (The specified module could not be found)) in Unknown on line 0

in my php.ini i just activate it
```
extension=sodium
extension=php_sodium
```
Any help ! please!
